### ISSUE TITLE
More discs

### ISSUE THREAD

**Issue by athensclub** [kithe-raker/gimmesong#59]
**More discs**
This feature will introduce different type of disc styles, including decorations on discs, and emoji in the center of the disc.

**DOD**
 - [x] Selectable list of discs when sending songs (show all discs that the user can use to send)
 - [x] Different types of discs is sent to the database according to the sender
 - [x] Different types of discs are displayed according to the disc sent by the sender (in my songs/playlist etc.)
~~Each users have their own discs inventory~~ <- (Not required in this version)

**Comment by athensclub** [comment]
**Selectable list of discs when sending songs**

![result1](https://i.imgur.com/GKa1J6z.png)
![result2](https://i.imgur.com/hVojxW7.png)

**Comment by athensclub** [comment]
As far as I know, the currently required feature includes selecting disc, and the disc will be displayed according to their sender's choice of disc and sharing. I've implemented selecting disc (hard-coded) and have commented on where to implement the api code.

**Comment by kithe-raker** [comment]
@athensclub In #65 I've implemented dynamic vinyl style in sent/added song data
The schema data will looks like something like [this](https://github.com/kithe-raker/gimmesong/issues/65#issuecomment-1356800883)
.
If u have any problem implementing with the vinyl render component, let me know.

**Comment by athensclub** [comment]
@kithe-raker I believe that I've implemented all the features in the UI part including: show discs that can be sent, pass the disc style to API when sending song, and render disc styles from the API. Waiting to test the API :)

**Comment by kithe-raker** [comment]
@athensclub  Found some weird behavior, make the vinyl component not render properly.
.
**here a step to reproduce**
1. open a song request(any playlist)
2. click `add song` button
3. choose a song
4. close the `add song` tab
5. click `add song` button again
6. choose a song
<img width="566" alt="ภาพถ่ายหน้าจอ 2565-12-20 เวลา 22 34 55" src="https://user-images.githubusercontent.com/54874488/208705048-845fcafc-dc84-458e-91aa-4d11a922d1df.png">

This issue can also achieved by open send song to someone, then click add song button in the playlist

**Comment by kithe-raker** [comment]
> @athensclub Found some weird behavior, make the vinyl component not render properly. . **here a step to reproduce**
> 
> 1. open a song request(any playlist)
> 2. click `add song` button
> 3. choose a song
> 4. close the `add song` tab
> 5. click `add song` button again
> 6. choose a song
> 
> <img alt="ภาพถ่ายหน้าจอ 2565-12-20 เวลา 22 36 38" width="526" src="https://user-images.githubusercontent.com/54874488/208705446-bde9d615-6e1c-40cc-a89c-ac931c1b8093.png">
> 
> This issue can also achieved by open send song to someone, then click add song button in the playlist

Oh, looks like the center icon that overlay the vinyl doesn't render for some reason.

**Comment by athensclub** [comment]
@kithe-raker 
After logging the query result using the code below, It seems that the SDK is returning the background and the center url. Not sure if that is the problem you are talking about? 

<img width="446" alt="image" src="https://user-images.githubusercontent.com/43963760/208730830-1debac07-3199-49a4-a6e9-dd81c0557053.png">

**Result from console**
```
{
    "background": [
        {
            "id": "id",
            "image_url": "/static/media/gimmesong_logo.1600816a01874bced6f6.png"
        }
    ],
    "center": [
        {
            "id": "id",
            "image_url": "/static/media/gimmesong_logo.1600816a01874bced6f6.png"
        }
    ]
}
```

**Comment by kithe-raker** [comment]
> @kithe-raker After logging the query result using the code below, It seems that the SDK is returning the background and the center url as the same value. Not sure if that is the problem you are talking about?
> 
> <img alt="image" width="446" src="https://user-images.githubusercontent.com/43963760/208730830-1debac07-3199-49a4-a6e9-dd81c0557053.png">
> 
> **Result from console**
> 
> ```
> {
>     "background": [
>         {
>             "id": "id",
>             "image_url": "/static/media/gimmesong_logo.1600816a01874bced6f6.png"
>         }
>     ],
>     "center": [
>         {
>             "id": "id",
>             "image_url": "/static/media/gimmesong_logo.1600816a01874bced6f6.png"
>         }
>     ]
> }
> ```
> 
> Since both background and center is the image below ![image](https://user-images.githubusercontent.com/43963760/208731566-097567a0-2eb9-4f83-b7f9-09d15517f908.png)
> 
> The frontend is rendering the disc as the background and also use the disc as a center (you can see a smaller, identical disc at the center of the disc)

yeah correct, but still, why it render and got a different result.
<img width="567" alt="ภาพถ่ายหน้าจอ 2565-12-21 เวลา 13 04 01" src="https://user-images.githubusercontent.com/54874488/208833169-24f632c1-b2be-4352-b2ea-0b8334ffea2a.png">

**Comment by kithe-raker** [comment]
nevermind, looks like its work now after update the vinyl style's object

**Comment by kithe-raker** [comment]
@athensclub I've just pushed it, can you try it out, and edit the fetch style function?
because it only got one style, it should have 3 style

**Comment by athensclub** [comment]
So, are we doing every combination of background and center? If not, how do I know which background or center belongs to which disc?

**Comment by kithe-raker** [comment]
> So, are we doing every combination of background and center? If not, how do I know which background or center belongs to which disc?

yeah, we only have one background. so u can do every combination of background and center

**Comment by athensclub** [comment]
There is also another issue that the discs that is sent before today will have vinyl style returned as undefined. See below:
<img width="223" alt="image" src="https://user-images.githubusercontent.com/43963760/208847357-3c581559-d9d2-4ceb-8b20-0e5d9a1e3ca3.png">

Do you want me to handle this in the frontend or will you handle it in the backend?

**Comment by athensclub** [comment]
I've also push the updated fetchDiscs function which now uses all combination of center and background.

**Comment by kithe-raker** [comment]
> I've also push the updated fetchDiscs function which now uses all combination of center and background.

Oh, actually I have to handle it in the sdk part, and I forget about it.
Working on it

**Comment by kithe-raker** [comment]
@athensclub fixed in 5e6ec4125edc7dd64052bffb56d8caf97eb207b0
